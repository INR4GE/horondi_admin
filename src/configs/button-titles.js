const buttonTitles = {
  DELETE_TITLE: 'Видалити',
  EDIT_TITLE: 'Редагувати',
  CREATE_NEWS_TITLE: 'Додати новину',
  CREATE_PATTERN_TITLE: 'Додати гобелен',
  REMOVE_TITLE: 'Видалити новину',
  PATTERN_REMOVE_TITLE: 'Видалити гобелен',
  REMOVE_BUSINESS_PAGE_TITLE: 'Видалити сторінку',
  CANCEL_TITLE: 'Відмінити',
  LOGOUT_TITLE: 'Вихід',
  CREATE_BUSINESS_PAGE: 'Додати бізнес сторінку',
  CREATE_CONTACT_TITLE: 'Додати контакти',
  REMOVE_CONTACT_TITLE: 'Видалити контакт',
  REMOVE_USER_TITLE: 'Видалити користувача',
  SWITCH_USER_STATUS_TITLE: 'Змінити статус користувача',
  USER_INACTIVE_TITLE: 'Деактивувати',
  USER_ACTIVE_TITLE: 'Активувати',
  ADD_CATEGORY: 'Додати категорію',
  GO_BACK_TITLE: 'Назад',
  DELETE_CATEGORY: 'Видалити категорію',
  ADD_SUBCATEGORY: 'Додати підкатегорію',
  ADD_CATEGORY_IMAGE: 'Зберегти посилання',
  ADD_CATEGORY_NAME: 'Додати назву',
  CANCEL: 'Відмінити',
  SAVE_CATEGORY: 'Зберегти категорію',
  SAVE_SUBCATEGORY: 'Зберегти підкатегорію',
  CREATE_SPECIAL_USER: 'Створити спецкористувача',
  CREATE_CATEGORY: 'Створити категорію',
  CREATE_SUBCATEGORY: 'Створити підкатегорію',
  USER_UNACTIVE_TITLE: 'Деактивувати',
  REMOVE_COMMENT_TITLE: 'Видалити коментар',
  SHOW_COMMENTS_TITLE: 'Переглянути коментарі',
  HIDE_COMMENTS_TITLE: 'Приховати коментарі',
  ADD_PHOTO_LABEL: '+',
  REMOVE_EMAIL_QUESTION: 'Видалити запитання?',

  titleGenerator: (editMode, isMain) => {
    const editModeMap = new Map([
      [true, 'Зберегти'],
      [false, 'Створити']
    ]);
    const isMainMap = new Map([
      [true, 'категорію'],
      [false, 'підкатегорію']
    ]);
    return `${editModeMap.get(editMode)} ${isMainMap.get(isMain)}`;
  }
};
export default buttonTitles;
