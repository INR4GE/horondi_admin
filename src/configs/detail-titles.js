const detailTitles = {
  users: {
    avatar: { id: 'avatar' },
    name: { id: 'name' },
    status: { id: 'status' },
    primarySection: [
      { id: 'country', label: 'Країна' },
      { id: 'city', label: 'Місто' }
    ],
    secondarySection: [
      { id: 'adress', label: 'Адреса' },
      { id: 'postCode', label: 'Поштовий індекс' }
    ]
  },
  emailQuestions: {
    customer: {
      date: 'Дата запитання:',
      status: 'Статус:',
      sender: 'Відправник:',
      email: 'Email:',
      question: 'Запитання:'
    },
    admin: {
      date: 'Дата відповіді:',
      admin: 'Адмін:',
      answer: 'Відповідь:'
    }
  }
};

export default detailTitles;
