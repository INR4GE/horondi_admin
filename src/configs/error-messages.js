export const errorMessages = {
  USER_NOT_FOUND: 'Користувач не знайдений!',
  USER_NOT_AUTHORIZED: 'Користувач не отримав прав доступу',
  INVALID_PERMISSIONS: 'Недостатньо прав користувача',
  WRONG_CREDENTIALS: 'Неправильно вказані вхідні дані',
  INPUT_NOT_VALID: 'Неправильні ввідні дані',
  USER_ALREADY_EXIST: 'Користувач з такими даними вже існує',
  INVALID_ADMIN_INVITATIONAL_TOKEN:
    'Неправильне посилання на створення користувача'
};

export const loginErrorMessages = {
  INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  ENTER_EMAIL_MESSAGE: 'Введіть email',
  PASSWORD_MIN_LENGTH_MESSAGE: 'Пароль повинен містити не менше 8 символів',
  PASSWORD_MAX_LENGTH_MESSAGE: 'Пароль повинен містити не більше 20 символів',
  PASSWORD_LANG_MESSAGE: 'Використовуйте латиницю різних регістрів та цифри',
  ENTER_FIRSTNAME_MESSAGE: "Введіть ім'я",
  ENTER_LASTNAME_MESSAGE: 'Введіть прізвище',
  ENTER_PASSWORD_MESSAGE: 'Введіть пароль',
  FIRSTNAME_MIN_LENGTH_MESSAGE: "Ім'я повинно містити не менше 2 символів",
  LASTNAME_MIN_LENGTH_MESSAGE: 'Прізвище повинно містити не менше 2 символів',
  FIRSTNAME_MAX_LENGTH_MESSAGE: "Ім'я повинно містити не більше 30 символів",
  LASTNAME_MAX_LENGTH_MESSAGE: 'Прізвище повинно містити не більше 30 символів',
  SELECT_ROLE_MESSAGE: 'Оберіть роль'
};
export const patternErrorMessages = {
  PATTERN_VALIDATION_ERROR: 'Мінімум 2 символи',
  PATTERN_ERROR_MESSAGE: 'Поле не може бути порожнім',
  PATTERN_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри'
};
export const newsErrorMessages = {
  NAME_MAX_LENGTH_MESSAGE: `Ім'я автора повинне містити не більше 100 символів`,
  NAME_MIN_LENGTH_MESSAGE: `Ім'я автора повинне містити не менше 6 символів`,
  TITLE_MAX_LENGTH_MESSAGE: 'Заголовок повинен містити не більше 100 символів',
  TITLE_MIN_LENGTH_MESSAGE: 'Заголовок повинен містити не менше 10 символів'
};
export const modelErrorMessages = {
  MODEL_VALIDATION_ERROR: 'Мінімум 2 символи',
  MODEL_ERROR_MESSAGE: 'Поле не може бути порожнім',
  MODEL_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри'
};
export const contactErrorMessages = {
  INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  ENTER_EMAIL_MESSAGE: 'Введіть email',
  PHONE_NUMBER_LENGTH_MESSAGE:
    'Довжина номеру телефону повинна містити 12 символів',
  PHONE_NUMBER_TYPE_MESSAGE: 'Номер повинен містити лише числа',
  ENTER_PHONE_NUMBER_MESSAGE: 'Введіть номер',
  INPUT_LENGTH_MESSAGE: 'Довжина повинна містити не менше 10 символів',
  ENTER_UK_SCHEDULE_MESSAGE: 'Введіть розклад українською',
  ENTER_EN_SCHEDULE_MESSAGE: 'Введіть розклад англійською',
  ENTER_UK_ADDRESS_MESSAGE: 'Введіть адресу українською',
  ENTER_EN_ADDRESS_MESSAGE: 'Введіть адресу англійською',
  IMAGE_FORMAT_MESSAGE:
    'Введіть коректний формат, наприклад: https://example.com/',
  ENTER_LINK_MESSAGE: 'Введіть посилання',
  SELECT_IMAGES_MESSAGE: 'Завантажте зображення для карт'
};

export const emailQuestionsErrorMessages = {
  ANSWER_INPUT_MESSAGE: 'Введіть текст для відповіді'
};

export const statsErrorMessages = {
  NO_STATS: 'Статистика для вибраного значення відсутня'
};
