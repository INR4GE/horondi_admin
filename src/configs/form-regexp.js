const formRegExp = {
  patternMaterial: '^[A-Za-z][A-Za-z0-9]*$',
  email:
    '^([\\w-]+(?:\\.[\\w-]+)*)@((?:[\\w-]+\\.)*\\w[\\w-]{0,66})\\.([a-z]{2,6}(?:\\.[a-z]{2})?)$',
  password: '^(?!.* )(?=.*[0-9])(?=.*[A-Z]).{8,30}$',
  unwrapHtml: /(<([^>]+)>)/gi,
  enAddressRegex: '^[A-Za-z0-9_|,| |./]+$',
  editorField: /^<p><br><\/p>$/
};
export default formRegExp;
