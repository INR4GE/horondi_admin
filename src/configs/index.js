import ImportContactsIcon from '@material-ui/icons/ImportContacts';
import BusinessCenterIcon from '@material-ui/icons/BusinessCenter';
import ImportLocationOnIcon from '@material-ui/icons/LocationOn';
import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket';
import CategoryIcon from '@material-ui/icons/Category';
import PeopleIcon from '@material-ui/icons/People';

export const routes = {
  pathToLogin: '/',
  pathToNews: '/news',
  pathToNewsDetails: '/news/:id',
  pathToAddNews: '/newsadd',
  pathToBusinessPages: '/business-pages',
  pathToAddBusinessPage: '/business-page-add',
  pathToBusinessPageDetails: '/business-pages/:id',
  pathToUsers: '/users',
  pathToUsersDetails: '/users/:id',
  pathToProducts: '/products',
  pathToAddProduct: '/add-product',
  pathToEditProduct: '/product/:id',
  pathToCategories: '/categories',
  pathToAddCategory: '/add-category',
  pathToEditCategory: '/add-category/:id',
  pathToContacts: '/contacts',
  pathToContactsEdit: '/contacts/:id',
  pathToAddContact: '/add-contact'
};

export const config = {
  app: {
    title: 'Horondi Admin Portal',
    menuCategories: [
      ['Продукти', routes.pathToProducts, ShoppingBasketIcon],
      ['Категорії', routes.pathToCategories, CategoryIcon],
      ['Користувачі', routes.pathToUsers, PeopleIcon],
      ['Бізнес сторінки', routes.pathToBusinessPages, BusinessCenterIcon],
      ['Новини', routes.pathToNews, ImportContactsIcon],
      ['Контакти', routes.pathToContacts, ImportLocationOnIcon]
    ],
    routes,
    serverUrl: 'http://localhost:5000/',
    drawerWidth: 220,
    snackBarDuration: 4000,
    rowsPerPageOptions: [10, 25, 50, 100]
  },
  languages: ['uk', 'en'],
  buttonStyles: {
    ACCEPT_BUTTON_STYLE: 'secondary'
  },
  templates: {
    categoryTemplate: {
      available: false,
      code: '',
      images: {
        large: '',
        medium: '',
        small: '',
        thumbnail: ''
      },
      isMain: false,
      name: [],
      subcategories: []
    }
  },
  tableHeadRowTitles: {
    news: ['Аватар', 'Автор', 'Заголовок', 'Дії'],
    businessPages: ['Аватар', 'Код', 'Заголовок', 'Дії'],
    products: [
      'Фото',
      'Назва',
      'Категорія',
      'Модель',
      'Гобелен',
      'Ціна(грн)',
      'Рейтинг',
      'Кількість покупок',
      'Дії'
    ],
    categories: ['№', 'Категорія', 'Дії'],
    subcategories: ['№', 'Підкатегорія', 'Доступна', 'Дії'],
    categoryName: ['№', 'Мова', 'Назва', 'Дії'],
    categoryImages: ['№', 'Розмір', 'Посилання', 'Дії'],
    users: ['Аватар', "Ім'я", 'Мобільний номер', 'Пошта', 'Статус', 'Дії'],
    contacts: ['Номер телефону', 'Email', 'Адреса', 'Дії']
  },
  detailTitles: {
    users: {
      avatar: { id: 'avatar' },
      name: { id: 'name' },
      status: { id: 'status' },
      primarySection: [
        { id: 'country', label: 'Країна' },
        { id: 'city', label: 'Місто' }
      ],
      secondarySection: [
        { id: 'adress', label: 'Адреса' },
        { id: 'postCode', label: 'Поштовий індекс' }
      ]
    }
  },
  tableSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'medium'
  },
  iconSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'default'
  },
  theme: {
    DARK_THEME: 'dark',
    LIGHT_THEME: 'light'
  },
  statuses: {
    SUCCESS_ADD_STATUS: 'Успішно додано!',
    SUCCESS_DELETE_STATUS: 'Успішно видалено!',
    SUCCESS_UPDATE_STATUS: 'Успішно змінено!',
    ERROR_PAGE_STATUS: 'Сторінку не знайдено!',
    USER_ACTIVE_STATUS: 'Активний(-a)',
    USER_INACTIVE_STATUS: 'Неактивний(-a)',
    LOGIN_PAGE_STATUS: 'Невірний логін або пароль'
  },
  errorMessages: {
    USER_NOT_FOUND: 'Користувач не знайдений!',
    USER_NOT_AUTHORIZED: 'Користувач не отримав прав доступу'
  },
  buttonTitles: {
    DELETE_TITLE: 'Видалити',
    EDIT_TITLE: 'Редагувати',
    CREATE_NEWS_TITLE: 'Додати новину',
    REMOVE_TITLE: 'Видалити новину',
    REMOVE_BUSINESS_PAGE_TITLE: 'Видалити сторінку',
    CANCEL_TITLE: 'Відмінити',
    LOGOUT_TITLE: 'Вихід',
    CREATE_BUSINESS_PAGE: 'Додати бізнес сторінку'
  },
  messages: {
    REMOVE_MESSAGE: 'Ви впевнені, що хочете видалити цю новину?',
    REMOVE_BUSINESS_PAGE: 'Ви впевнені, що хочете видалити цю сторінку?',
    LOGOUT_MESSAGE: 'Ви впевнені, що хочете вийти?',
    CREATE_CONTACT_TITLE: 'Додати контакти',
    REMOVE_CONTACT_TITLE: 'Видалити контакт',
    REMOVE_USER_TITLE: 'Видалити користувача',
    SWITCH_USER_STATUS_TITLE: 'Змінити статус користувача',
    CANCEL_TITLE: 'Відмінити',
    LOGOUT_TITLE: 'Вихід',
    ADD_CATEGORY: 'Додати категорію',
    DELETE_CATEGORY: 'Видалити категорію',
    ADD_SUBCATEGORY: 'Додати підкатегорію',
    ADD_CATEGORY_IMAGE: 'Зберегти посилання',
    ADD_CATEGORY_NAME: 'Додати назву',
    CANCEL: 'Відмінити',
    SAVE_CATEGORY: 'Зберегти категорію',
    SAVE_SUBCATEGORY: 'Зберегти підкатегорію',
    CREATE_CATEGORY: 'Створити категорію',
    CREATE_SUBCATEGORY: 'Створити підкатегорію',
    titleGenerator: (editMode, isMain) => {
      const editModeMap = new Map([
        [true, 'Зберегти'],
        [false, 'Створити']
      ]);
      const isMainMap = new Map([
        [true, 'категорію'],
        [false, 'підкатегорію']
      ]);
      return `${editModeMap.get(editMode)} ${isMainMap.get(isMain)}`;
    },
    GO_BACK_TITLE: 'Назад'
  },
  formRegExp: {
    email:
      '^([\\w-]+(?:\\.[\\w-]+)*)@((?:[\\w-]+\\.)*\\w[\\w-]{0,66})\\.([a-z]{2,6}(?:\\.[a-z]{2})?)$',
    password: '^(?!.* )(?=.*[0-9])(?=.*[A-Z]).{8,30}$',
    unwrapHtml: /(<([^>]+)>)/gi,
    enAddressRegex: '^[A-Za-z0-9_|,| |./]+$',
    editorField: /^<p><br><\/p>$/
  },
  loginErrorMessages: {
    INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
    ENTER_EMAIL_MESSAGE: 'Введіть email',
    PASSWORD_MIN_LENGTH_MESSAGE: 'Пароль повинен містити не менше 8 символів',
    PASSWORD_LANG_MESSAGE: 'Використовуйте латиницю різних регістрів та цифри',
    ENTER_PASSWORD_MESSAGE: 'Введіть пароль'
  },
  newsErrorMessages: {
    NAME_MAX_LENGTH_MESSAGE: `Ім'я автора повинне містити не більше 100 символів`,
    NAME_MIN_LENGTH_MESSAGE: `Ім'я автора повинне містити не менше 6 символів`,
    TITLE_MAX_LENGTH_MESSAGE:
      'Заголовок повинен містити не більше 100 символів',
    TITLE_MIN_LENGTH_MESSAGE: 'Заголовок повинен містити не менше 10 символів'
  },
  contactErrorMessages: {
    INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
    ENTER_EMAIL_MESSAGE: 'Введіть email',
    PHONE_NUMBER_LENGTH_MESSAGE:
      'Довжина номеру телефону повинна містити 12 символів',
    PHONE_NUMBER_TYPE_MESSAGE: 'Номер повинен містити лише числа',
    ENTER_PHONE_NUMBER_MESSAGE: 'Введіть номер',
    INPUT_LENGTH_MESSAGE: 'Довжина повинна містити не менше 10 символів',
    ENTER_UK_SCHEDULE_MESSAGE: 'Введіть розклад українською',
    ENTER_EN_SCHEDULE_MESSAGE: 'Введіть розклад англійською',
    ENTER_UK_ADDRESS_MESSAGE: 'Введіть адресу українською',
    ENTER_EN_ADDRESS_MESSAGE: 'Введіть адресу англійською',
    IMAGE_FORMAT_MESSAGE:
      'Введіть коректний формат, наприклад: https://example.com/',
    ENTER_LINK_MESSAGE: 'Введіть посилання'
  },
  paginationPayload: {
    skip: 0,
    limit: 5,
    newsPerPage: 6
  },
  contactsPaginationPayload: {
    skip: 0,
    limit: 6,
    contactsPerPage: 7
  },
  product: {
    sortBySelectOptions: [
      {
        label: 'популярністю',
        value: 'popularity'
      },
      {
        label: 'від дорогих до дешевих',
        value: 'sortDesc'
      },
      {
        label: 'від дешевих до дорогих',
        value: 'sortAsc'
      },
      {
        label: 'рейтингом',
        value: 'rate'
      }
    ],
    stepsLabels: [
      'Введіть інформацію про продукт',
      'Оберіть категорію, підкатегорію, модель, колір, гобелен та ціну продукту',
      'Оберіть опційні параметри',
      'Завантажте фото для продукту',
      'Підтвердження створення продукту'
    ],
    infoLabels: [
      { label: 'Назва', name: 'name', required: true },
      { label: 'Основний матеріал', name: 'mainMaterial', required: true },
      { label: 'Внутрішній матеріал', name: 'innerMaterial', required: false },
      { label: 'Замок', name: 'closure', required: false },
      { label: 'Опис', name: 'description', required: false }
    ],
    selectsLabels: [
      { label: 'Категорія ', name: 'category', type: 'select', required: true },
      { label: 'Підкатегорія ', name: 'subcategory', type: 'select', required: true },
      { label: 'Модель ', name: 'model', type: 'select', required: true },
      { label: 'Колір ', name: 'colors', type: 'select', required: true },
      { label: 'Гобелен ', name: 'pattern', type: 'select', required: true },
      { label: 'Ціна(USD) ', name: 'basePrice', type: 'number', required: true },
      { label: 'Довжина лямок(см) ', name: 'strapLengthInCm', type: 'number', required: false }
    ],
    responsive: {
      superLargeDesktop: {
        breakpoint: { max: 4000, min: 3000 },
        items: 1
      },
      desktop: {
        breakpoint: { max: 3000, min: 1024 },
        items: 1
      },
      tablet: {
        breakpoint: { max: 1146, min: 464 },
        items: 1
      },
      mobile: {
        breakpoint: { max: 810, min: 0 },
        items: 1
      }
    },
    optionsLabels: [
      { label: 'Розміри', name: 'sizes' },
      { label: 'Нижні матеріали', name: 'bottomMaterials' }
    ],
    sizeCardsLabels: [
      { label: 'Розмір', name: 'name' },
      { label: `Об'єм(л)`, name: 'volumeInLiters' },
      { label: 'Ширина(см)', name: 'widthInCm' },
      { label: 'Висота(см)', name: 'heightInCm' },
      { label: 'Глибина(см)', name: 'depthInCm' }
    ],
    materialsLabels: [{ label: `Назва`, name: 'name' }],
    optionsValues: {
      sizes: [],
      bottomMaterials: [],
      additions: false
    }
  },
  popularity: 'popularity',
  rate: 'rate',
  sortAsc: 'sortAsc',
  sortDesc: 'sortDesc',
  submitKey: 'Enter',
  selectedLanguages: {
    uk: {
      name: 'uk',
      checked: false
    },
    en: {
      name: 'en',
      checked: false
    }
  },
  initialLanguageValues: [
    { lang: 'uk', value: '' },
    { lang: 'en', value: '' }
  ],
  imagePrefix: 'https://horondi.blob.core.windows.net/horondi/images/'
};
