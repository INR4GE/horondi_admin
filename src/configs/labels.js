const labels = {
  model: {
    name: 'Назва',
    description: 'Опис',
    image: 'Фото моделі',
    category: 'Категорія',
    available: 'Доступний',
    show: 'Показати',
    avatarText: 'Фото',
    priority: 'Пріорітет',
    showEnable: 'Так',
    showDisable: 'Ні'
  },
  pattern: {
    image: 'Фото гобелена',
    material: 'Код матеріалу',
    available: 'Доступний',
    handmade: 'Зроблений вручну',
    avatarText: 'Фото'
  },
  product: {
    sortBySelectOptions: [
      {
        label: 'популярністю',
        value: 'popularity'
      },
      {
        label: 'від дорогих до дешевих',
        value: 'sortDesc'
      },
      {
        label: 'від дешевих до дорогих',
        value: 'sortAsc'
      },
      {
        label: 'рейтингом',
        value: 'rate'
      }
    ],
    stepsLabels: [
      'Введіть інформацію про продукт',
      'Оберіть категорію, підкатегорію, модель, колір, гобелен та ціну продукту',
      'Вкажіть ціну продукту',
      'Оберіть опційні параметри',
      'Завантажте фото для продукту',
      'Підтвердження створення продукту'
    ],
    infoLabels: [
      { label: 'Назва', name: 'name', required: true },
      {
        label: 'Основний матеріал',
        name: 'mainMaterial',
        required: true
      },
      {
        label: 'Внутрішній матеріал',
        name: 'innerMaterial',
        required: false
      },
      { label: 'Замок', name: 'closure', required: false },
      {
        label: 'Довжина лямок(см)',
        name: 'strapLengthInCm',
        required: false
      },
      { label: 'Опис', name: 'description', required: false }
    ],
    selectsLabels: [
      {
        label: 'Категорія ',
        name: 'category',
        type: 'select',
        required: true
      },
      {
        label: 'Підкатегорія ',
        name: 'subcategory',
        type: 'select',
        required: true
      },
      {
        label: 'Модель ',
        name: 'model',
        type: 'select',
        required: true
      },
      {
        label: 'Колір ',
        name: 'colors',
        type: 'select',
        required: true
      },
      {
        label: 'Гобелен ',
        name: 'pattern',
        type: 'select',
        required: true
      }
    ],
    optionsLabels: [
      { label: 'Розміри', name: 'sizes' },
      { label: 'Нижні матеріали', name: 'bottomMaterials' }
    ],
    sizeCardsLabels: [
      { label: 'Розмір', name: 'name' },
      { label: `Об'єм(л)`, name: 'volumeInLiters' },
      { label: 'Ширина(см)', name: 'widthInCm' },
      { label: 'Висота(см)', name: 'heightInCm' },
      { label: 'Глибина(см)', name: 'depthInCm' }
    ],
    materialsLabels: [{ label: `Назва`, name: 'name' }],
    optionsValues: {
      sizes: [],
      bottomMaterials: [],
      additions: false
    },
    priceLabel: {
      label: 'Ціна(USD) ',
      name: 'basePrice'
    }
  },
  doughnut: {
    select: [
      { label: 'Популярні категорії', value: 'categories' },
      { label: 'Останні замовлення', value: 'orders' }
    ],
    dateMenuOptions: [
      { label: 'За 7 Днів', value: 7 },
      { label: 'За 14 Днів', value: 14 },
      { label: 'За 30 Днів', value: 30 }
    ]
  },
  bar: {
    select: [
      { label: 'Популярні продукти', value: 'products' },
      { label: 'Успішно продані замовлення', value: 'orders' },
      { label: 'Останні зареєстровані користувачі', value: 'users' }
    ],
    descriptions: {
      users: 'Цей день',
      orders: 'Цей день',
      products: 'Кількість покупок'
    }
  }
};
export default labels;
