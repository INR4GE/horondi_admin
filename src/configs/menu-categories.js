import BusinessCenterIcon from '@material-ui/icons/BusinessCenter';
import CategoryIcon from '@material-ui/icons/Category';
import ImportContactsIcon from '@material-ui/icons/ImportContacts';
import ImportLocationOnIcon from '@material-ui/icons/LocationOn';
import PaletteIcon from '@material-ui/icons/Palette';
import PeopleIcon from '@material-ui/icons/People';
import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket';
import LiveHelpIcon from '@material-ui/icons/LiveHelp';
import SmsIcon from '@material-ui/icons/Sms';
import StyleIcon from '@material-ui/icons/Style';
import routes from './routes';

const menuCategories = [
  ['Продукти', routes.pathToProducts, ShoppingBasketIcon],
  ['Категорії', routes.pathToCategories, CategoryIcon],
  ['Користувачі', routes.pathToUsers, PeopleIcon],
  ['Бізнес сторінки', routes.pathToBusinessPages, BusinessCenterIcon],
  ['Останні коментарі', routes.pathToComments, SmsIcon],
  ['Контакти', routes.pathToContacts, ImportLocationOnIcon],
  ['Гобелени', routes.pathToPatterns, PaletteIcon],
  ['Новини', routes.pathToNews, ImportContactsIcon],
  ['Запитання & Відповіді', routes.pathToEmailQuestions, LiveHelpIcon],
  ['Моделі', routes.pathToModels, StyleIcon]
];
export default menuCategories;
