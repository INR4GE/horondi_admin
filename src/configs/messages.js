const messages = {
  REMOVE_MESSAGE: 'Ви впевнені, що хочете видалити цю новину?',
  REMOVE_BUSINESS_PAGE: 'Ви впевнені, що хочете видалити цю сторінку?',
  LOGOUT_MESSAGE: 'Ви впевнені, що хочете вийти?',
  DELETE_CATEGORY_MESSAGE: 'Ви впевнені, що хочете видалити цю категорію?',
  REMOVE_USER_MESSAGE: 'Ви впевнені,що хочете видалити цього користувача?',
  SWITCH_USER_STATUS_MESSAGE:
    'Ви впевнені,що хочете змінити статус користувача?',
  REMOVE_CONTACT_MESSAGE: 'Ви впевнені,що хочете видалити цей контакт?',
  REMOVE_COMMENT_MESSAGE: 'Ви впевнені, що хочете видалити цей коментар?',
  NO_COMMENTS_MESSAGE: 'Коментарі відсутні',
  PATTERN_REMOVE_MESSAGE: 'Ви впевнені, що хочете видалити цей гобелен?',
  EMAIL_QUESTION_REMOVE_MESSAGE:
    'Ви впевнені, що хочете видалити це запитання?',
  EMAIL_QUESTIONS_MOVE_TO_SPAM_MESSAGE:
    'Ви впевнені, що хочете перемістити вибрані запитання у СПАМ?',
  EMAIL_QUESTIONS_MOVE_TO_DELETE_MESSAGE:
    'Ви впевнені, що хочете видалити всі вибрані запитання?',
  EMAIL_QUESTION_SPAM_DETAILS:
    'Відповідь не доступна для запитань відзначених як "СПАМ"',
  MODEL_REMOVE_MESSAGE: 'Ви впевнені, що хочете видалити цю модель?',
  ERROR: 'ПОМИЛКА',
  EMPTY_LIST: 'Список порожній'
};
export default messages;
