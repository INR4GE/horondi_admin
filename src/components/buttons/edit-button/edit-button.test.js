import React from 'react';
import { render } from '@testing-library/react';
import EditButton from './edit-button';
import { config, inputTypes } from '../../../configs';
import '@testing-library/jest-dom';

const { EDIT_TITLE } = config.buttonTitles;

describe('delete button tests', () => {
  const mockCallBack = jest.fn();
  const wrapper = render(
    <EditButton size='small' onClickHandler={mockCallBack} />
  );

  it('should exist and have title', () => {
    const component = wrapper.getByTitle(EDIT_TITLE);
    expect(component).toMatchSnapshot();
    expect(component).toBeDefined();
    expect(component.title).toEqual(EDIT_TITLE);
    expect(component.type).toEqual(inputTypes.button);
    expect(component).not.toBeNull();
  });
});
