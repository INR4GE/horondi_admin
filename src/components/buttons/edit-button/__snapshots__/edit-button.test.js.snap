// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete button tests should exist and have title 1`] = `
<button
  aria-label="Редагувати"
  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
  tabindex="0"
  title="Редагувати"
  type="button"
>
  <span
    class="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
      />
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
